app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OffboardRouterActivity.kt

```diff
@@ -0,0 +1,257 @@
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun calculateRoute() {
+        locationEngine.getLastLocation(object : LocationEngineCallback<LocationEngineResult> {
+            override fun onSuccess(result: LocationEngineResult) {
+                findRouteWith(result)
+            }
+
+            override fun onFailure(exception: Exception) {
+                Timber.e(exception)
+            }
+        })
+    }
+
+    private fun findRouteWith(result: LocationEngineResult) {
+        val userLocation = result.lastLocation
+        if (userLocation == null) {
+            Timber.d("calculateRoute: User location is null, therefore, origin can't be set.")
+            return
+        }
+        val origin = Point.fromLngLat(userLocation.longitude, userLocation.latitude)
+        destination?.let { destinationPoint ->
+            if (TurfMeasurement.distance(origin, destinationPoint, TurfConstants.UNIT_METERS) < 50) {
+                return
+            }
+            val waypoints = mutableListOf(waypoint).filterNotNull()
+            val offboardRouter = MapboxOffboardRouter(this, Utils.getMapboxAccessToken(this))
+            directionsSession = MapboxDirectionsSession(
+                offboardRouter,
+                origin,
+                waypoints,
+                destinationPoint,
+                this
+            )
+            directionsSession?.requestRoutes()
+        }
+    }
+
+    /*
+     * DirectionSessions.RouteObserver
+     */
+
+    override fun onRoutesChanged(routes: List<Route>) {
+        routes.firstOrNull()?.let {
+            this.route = mapRouteToDirectionsRoute(it)
+            navigationMapRoute.addRoute(this.route)
+        }
+    }
+
+    override fun onRoutesRequested() {
+        Timber.d("onRoutesRequested: navigation.getRoute()")
+    }
+
+    override fun onRoutesRequestFailure(throwable: Throwable) {
+        Timber.e(throwable, "onRoutesRequestFailure: navigation.getRoute()")
+    }
+
+    /*
+     * Activity lifecycle methods
+     */
+
+    public override fun onResume() {
+        super.onResume()
+        mapView.onResume()
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        mapView.onPause()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapView.onStop()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        directionsSession?.cancel()
+        navigation.onDestroy()
+        mapboxMap?.removeOnMapClickListener(this)
+        mapView.onDestroy()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        mapView.onSaveInstanceState(outState)
+    }
+
+    override fun onMetricUpdated(@MetricEvent.Metric metric: String, jsonStringData: String) {
```

Comment:
Same as commented above re: is this being used in this example? We're not navigating and therefore we're not sending any events.